eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,154 @@
+# 3. If there is 'element-list' in javadoc jar, then copy to destination ${package}.txt.
+# 4. If no 'element-list', then parse the 'overview-frame.html' for the namespaces and copy to destination.
+# 5. If no 'overview-frame.html', then read folder 'com/azure/...' for namespaces. E.g some mgmt packages use this structure.
+# 6. Otherwise, return empty children.
+function Get-Toc-Children($package, $groupId, $version, $docRepoLocation, $isPreview) {
+    # Looking for the txt
+    $folder = "latest" 
+    if ($isPreview) {
+        $folder = "preview"
+    } 
+    $filePath = Join-Path "$docRepoLocation/metadata/$folder" "$package.txt"
+    if (!(Test-Path $filePath)) {
+        # Download from maven
+        # javadoc jar url. e.g.: https://repo1.maven.org/maven2/com/azure/azure-core/1.25.0/azure-core-1.25.0-javadoc.jar
+        $groupId = "$groupId" -replace "\.", "/"
+        $mavenJavadocJarUrl = "$packageDownloadUrl/$groupId/$package/$version/$package-$version-javadoc.jar"
+        Write-Host "Namespaces Not found. Downloading from maven repository $mavenJavadocJarUrl"
+        # A temp folder
+        $tempDirectory = Join-Path ([System.IO.Path]::GetTempPath()) "javadoc"
+        if (!(Test-Path $tempDirectory)) {
+            New-Item $tempDirectory -ItemType Directory
+        } 
+        try {
+            $javadocLocation = "$tempDirectory/$package-$version-javadoc.jar"
+            Invoke-WebRequest -Uri $mavenJavadocJarUrl -OutFile $javadocLocation
+            Write-Host "Download complete."
+        }
+        catch {
+            Write-Error "Not able to download javadoc jar from $mavenJavadocJarUrl."
+            return @()
+        }
+        Fetch-Namespaces-From-Javadoc -jarFilePath $javadocLocation -destination $filePath
+    }
+
+    if (!(Test-Path $filePath)) {
+        # Log and warn
+        Write-Host "Not able to find namespaces from javadoc jar $package-$version-javadoc.jar"
+    }
+    return (Get-Content $filePath | ForEach-Object {$_.Trim()})
+}
+  
+function Fetch-Namespaces-From-Javadoc ($jarFilePath, $tempLocation, $destination) {
+    $tempLocation = (Join-Path ([System.IO.Path]::GetTempPath()) "jarFiles")
+    if (Test-Path $tempLocation) {
+        Remove-Item $tempLocation/* -Recurse -Force
+    }
+    else {
+        New-Item -ItemType Directory -Path $tempLocation -Force
+    }
+
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($jarFilePath, $tempLocation)
+    if (Test-Path "$tempLocation/element-list") {
+        # Rename and move to location
+        Write-Host "Copying the element-list to $destination..."
+        Copy-Item "$tempLocation/element-list" -Destination $destination
+    }
+    elseif (Test-Path "$tempLocation/overview-frame.html") {
+        Parse-Overview-Frame -filePath "$tempLocation/overview-frame.html" -destination $destination
+    }
+    elseif (Test-Path "$tempLocation/com") {
+        Push-Location $tempLocation
```

Comment:
We've avoided using `Push-Location` and `Pop-Location` instead favoring something like: 

```pwsh
$originalLocation = Get-Location 
try { 
  Set-Location $tempLocation 
  ... 
} finally { 
  Set-Location $originalLocation
}
```

If any of the steps between `Push-Location` and `Pop-Location` throw an exception you might end up in an unknown directory. It's safer to be more defensive here.